packages/client/hooks/useQueryLoaderNow.tsx

```diff
@@ -1,23 +1,48 @@
 import areEqual from 'fbjs/lib/areEqual'
-import {useEffect, useRef} from 'react'
-import {PreloadableConcreteRequest, PreloadFetchPolicy, useQueryLoader} from 'react-relay'
+import {useCallback, useEffect, useRef} from 'react'
+import {
+  fetchQuery,
+  PreloadableConcreteRequest,
+  PreloadFetchPolicy,
+  useQueryLoader,
+  useRelayEnvironment
+} from 'react-relay'
 import {GraphQLTaggedNode, OperationType, VariablesOf} from 'relay-runtime'
 import useAtmosphere from './useAtmosphere'
 
 const useQueryLoaderNow = <TQuery extends OperationType>(
   preloadableRequest: GraphQLTaggedNode | PreloadableConcreteRequest<TQuery>,
   variables: VariablesOf<TQuery> = {},
-  fetchPolicy?: PreloadFetchPolicy
+  fetchPolicy?: PreloadFetchPolicy,
+  preventSuspense?: boolean
 ) => {
   const [queryRef, loadQuery] = useQueryLoader<TQuery>(preloadableRequest)
   const varRef = useRef(variables)
   const atmosphere = useAtmosphere()
+  const environment = useRelayEnvironment()
   if (!areEqual(variables, varRef.current)) {
     varRef.current = variables
   }
+
+  const refreshQuery = useCallback(() => {
```

Comment:
can you give some more detail here? if i set preventSuspense to false what bad things happen? any steps to reproduce the broken behavior?

Reply:
I merged in @nickoferrall 's code from [here](https://github.com/ParabolInc/parabol/pull/5650#discussion_r747732466)

Reply:
This prevented the flickering bug shown in the gif in this comment: https://github.com/ParabolInc/parabol/pull/5469#pullrequestreview-782585609

To test:

- set preventSuspense to false
- type in the search query
- notice that after each keystroke, the modal flashes

This is because the Spotlight Results suspend as the query executes. I followed [this pattern](https://relay.dev/docs/next/guided-tour/refetching/refreshing-queries/#if-you-need-to-avoid-suspense) from the Relay docs to prevent this from happening.

Reply:
ohhhhhh! i get it. it's like a workaround for `useTransition`. neat!